internal/btf/types.go

```diff
@@ -466,7 +477,8 @@ func inflateRawTypes(rawTypes []rawType, rawStrings stringTable) (namedTypes map
 
 		switch raw.Kind() {
 		case kindInt:
-			typ = &Int{id, name, raw.Size()}
+			encoding, offset, bits := raw.IntEncoding()
```

Comment:
I think this is incorrect: you're decoding BTF_INT_ENCODING, etc. from btf_type.info instead of the `u32` that follows btf_type. This has to use `raw.data.(uint32)` instead.

Reply:
I think it'd be good if you could add a small test for this. You should be able to extend the test that parses `vmlinux-btf.gz` to call `FindType` to find a bitset (`struct iphdr` comes to mind https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/ip.h#L86).

Reply:
Good catch! Fixed and added the test requested.